src/dos/drive_local.cpp

```diff
@@ -56,22 +56,50 @@ bool localDrive::FileCreate(DOS_File * * file,char * name,uint16_t /*attributes*
 	/* Test if file exists (so we need to truncate it). don't add to dirCache then */
 	bool existing_file = false;
 	
-	FILE * test = fopen_wrap(temp_name,"rb+");
+	FILE* test = fopen_wrap(temp_name, "rb");
 	if (test) {
 		fclose(test);
-		existing_file=true;
+		existing_file = true;
+	}
+
+	FILE* fp = nullptr;
 
+#ifdef WIN32
+	DWORD win_attribs = attributes ? (DWORD)attributes : FILE_ATTRIBUTE_NORMAL;
+	HANDLE win_fhandle = CreateFile(newname,
+	                                GENERIC_READ | GENERIC_WRITE,
+	                                0,
+	                                nullptr,
+	                                CREATE_ALWAYS,
+	                                win_attribs,
+	                                nullptr);
+	if (win_fhandle != INVALID_HANDLE_VALUE) {
+		int fd = _open_osfhandle((intptr_t)win_fhandle, _O_RDWR | _O_BINARY);
+		fp = _fdopen(fd, "wb+");
```

Comment:
Does this mean the Hidden, System and Archive attributes are supported now on Windows using NTFS? If so, I actually have quite mixed feelings about supporting all attributes on Windows, but only a subset on *nixes. 

I'd much prefer a consistent cross-platform behaviour and go with the least common denominator, otherwise we'll end up with scenarios where some programs works on Windows only, or has subtle bugs on other OSes than Windows.

Reply:
The current behavior is consistent with `localDrive::SetFileAttr`. The reason Hidden and System are not supported on Linux is simply because they _do not exist_ there. Linux does have a hidden files feature but it's used by prefixing filename with a dot.